streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -687,12 +687,17 @@ public StreamsConfig(final Map<?, ?> props) {
 
     private Map<String, Object> getCommonConsumerConfigs() {
         final Map<String, Object> clientProvidedProps = getClientPropsWithPrefix(CONSUMER_PREFIX, ConsumerConfig.configNames());
+        final Map<String, Object> clientProvidedCustomProps = getClientCustomProps(CONSUMER_PREFIX, PRODUCER_PREFIX);
 
         checkIfUnexpectedUserSpecifiedConsumerConfig(clientProvidedProps, NON_CONFIGURABLE_CONSUMER_DEFAULT_CONFIGS);
         checkIfUnexpectedUserSpecifiedConsumerConfig(clientProvidedProps, NON_CONFIGURABLE_CONSUMER_EOS_CONFIGS);
 
+        checkIfUnexpectedUserSpecifiedConsumerConfig(clientProvidedCustomProps, NON_CONFIGURABLE_CONSUMER_DEFAULT_CONFIGS);
+        checkIfUnexpectedUserSpecifiedConsumerConfig(clientProvidedCustomProps, NON_CONFIGURABLE_CONSUMER_EOS_CONFIGS);
+
         final Map<String, Object> consumerProps = new HashMap<>(eosEnabled ? CONSUMER_EOS_OVERRIDES : CONSUMER_DEFAULT_OVERRIDES);
         consumerProps.putAll(clientProvidedProps);
+        consumerProps.putAll(clientProvidedCustomProps);
```

Comment:
I think, we should add those before the other, to make sure that `consumer.myConfig` overwrites `myConfig`. We should add a test for this case, too.

Reply:
Nice one. Thanks. I'll change the order and add a test for that.